error installing llvm package on mac

- issue: `opam intall llvm` failed with error `ld: library not found for -lLLVM-13.`
- system info (Macbook with M1 pro chip): Darwin 21.0.1 Darwin Kernel Version 21.0.1
- `llvm-config --libs`
![image](https://user-images.githubusercontent.com/25857014/141659938-8dd8d9c5-1cbe-41c7-9776-e5b1a898e0e9.png)


![image](https://user-images.githubusercontent.com/25857014/141659767-ba03bee7-cffc-453b-841b-195faf32f742.png)

Using an older version while latest is supposed to be working isn't a solution.
Neither is using a different operating system while macOS is supposed to be supported.
This issue has stalled my workflow.
By copying libLLVM-13.dylib to /usr/local/lib, I have managed to bypass this particular error, but it still is failing.
```
[ERROR] The installation of llvm failed at "bash -ex install.sh llvm-config /Users/visualdevel/.opam/default/lib cmake make".

#=== ERROR while installing llvm.13.0.0 =======================================#
# context     2.1.2 | macos/x86_64 | ocaml.4.13.1 | https://opam.ocaml.org#788ed58b
# path        ~/.opam/default/.opam-switch/build/llvm.13.0.0
# command     ~/.opam/opam-init/hooks/sandbox.sh install bash -ex install.sh llvm-config /Users/visualdevel/.opam/default/lib cmake make
# exit-code   2
# env-file    ~/.opam/log/llvm-20491-6691dd.env
# output-file ~/.opam/log/llvm-20491-6691dd.out
### output ###
# [...]
# Copying OCaml library component llvm.cmxa to intermediate area
# Copying OCaml library component llvm.a to intermediate area
# Copying OCaml library component llvm.cmi to intermediate area
# Copying OCaml library component llvm.cmti to intermediate area
# Copying OCaml library component llvm.cmt to intermediate area
# Copying OCaml library component llvm.cmx to intermediate area
# Copying OCaml library component dllllvm.dylib to intermediate area
# Error copying file "/Users/visualdevel/.opam/default/.opam-switch/build/llvm.13.0.0/build/bindings/ocaml/llvm/dllllvm.dylib" to "/Users/visualdevel/.opam/default/.opam-switch/build/llvm.13.0.0/build/./lib/ocaml/llvm/".
# make[3]: *** [ocaml_llvm] Error 1
# make[2]: *** [bindings/ocaml/llvm/CMakeFiles/ocaml_llvm.dir/all] Error 2
# make[1]: *** [CMakeFiles/ocaml_all.dir/rule] Error 2
# make: *** [ocaml_all] Error 2
```
Please fix this. I'll be glad to help test solutions.
I would fix this by myself, but I have no idea where the error is coming from.

--- a/packages/llvm/llvm.10.0.0/files/fix-macos.patch
+++ b/packages/llvm/llvm.10.0.0/files/fix-macos.patch
@@ -1,0 +1,30 @@
+From 98ca3c8802d0ae92a5baf34ffdc454d680276845 Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sat, 25 Dec 2021 18:21:57 +0000
+Subject: [PATCH 2/2] Fix support for building the OCaml binding on macOS
+
+The output of ocamlmklib is always .a and .so
+---
+ llvm/cmake/modules/AddOCaml.cmake | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index b7e2ba430344..dc80abcc6e75 100644
+--- a/cmake/modules/AddOCaml.cmake
++++ b/cmake/modules/AddOCaml.cmake
+@@ -40,10 +40,10 @@ function(add_ocaml_library name)
+   set(ocaml_outputs "${bin}/${name}.cma")
+   if( ARG_C )
+     list(APPEND ocaml_outputs
+-         "${bin}/lib${name}${CMAKE_STATIC_LIBRARY_SUFFIX}")
++         "${bin}/lib${name}.a")
+     if ( BUILD_SHARED_LIBS )
+       list(APPEND ocaml_outputs
+-           "${bin}/dll${name}${CMAKE_SHARED_LIBRARY_SUFFIX}")
++           "${bin}/dll${name}.so")
+     endif()
+   endif()
+   if( HAVE_OCAMLOPT )
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.10.0.0/files/fix-rhel.patch
+++ b/packages/llvm/llvm.10.0.0/files/fix-rhel.patch
@@ -1,0 +1,45 @@
+From 58e8402f9000d0dfe67b3df99ed8cd21b8325acd Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sun, 26 Dec 2021 05:09:10 +0000
+Subject: [PATCH] Handle RHEL-based distributions (do not have
+ libLLVM-<VERSION>.so, only libLLVM.so)
+
+---
+ llvm/cmake/modules/AddOCaml.cmake | 12 +++++++-----
+ 1 file changed, 7 insertions(+), 5 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index dc80abcc6e75..f51a339f084b 100644
+--- a/cmake/modules/AddOCaml.cmake
++++ b/cmake/modules/AddOCaml.cmake
+@@ -66,20 +66,22 @@ function(add_ocaml_library name)
+     list(APPEND ocaml_flags "-custom")
+   endif()
+ 
++  if(LLVM_OCAML_OUT_OF_TREE)
++    list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
++  endif()
++
+   if(LLVM_LINK_LLVM_DYLIB)
+     list(APPEND ocaml_flags "-lLLVM")
+   else()
+     explicit_map_components_to_libraries(llvm_libs ${ARG_LLVM})
+-    if( BUILD_SHARED_LIBS )
+-      list(APPEND ocaml_flags "-lLLVM-${LLVM_VERSION_MAJOR}${LLVM_VERSION_SUFFIX}")
++
++    if(BUILD_SHARED_LIBS AND LLVM_OCAML_OUT_OF_TREE)
++      list(APPEND ocaml_flags ${LLVM_OCAML_EXTERNAL_LLVM_LIBS})
+     else()
+       foreach( llvm_lib ${llvm_libs} )
+         list(APPEND ocaml_flags "-l${llvm_lib}" )
+       endforeach()
+     endif()
+-    if(LLVM_OCAML_OUT_OF_TREE)
+-      list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
+-    endif()
+ 
+     get_property(system_libs TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
+     foreach(system_lib ${system_libs})
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.10.0.0/files/fix-shared.patch
+++ b/packages/llvm/llvm.10.0.0/files/fix-shared.patch
@@ -1,5 +1,15 @@
-diff --git a/cmake/modules/AddOCaml.cmake b/cmake/modules/AddOCaml.cmake
-index 554046b20..b27cbd36c 100644
+From 7eaf48ca7f0b2ba0b5871b48fbe4a1b1460afbef Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sat, 25 Dec 2021 18:21:00 +0000
+Subject: [PATCH 1/2] Add support for out-of-tree shared linked OCaml LLVM
+ binding
+
+---
+ llvm/cmake/modules/AddOCaml.cmake | 13 ++++++++++---
+ 1 file changed, 10 insertions(+), 3 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index 891c9e6d618c..b7e2ba430344 100644
 --- a/cmake/modules/AddOCaml.cmake
 +++ b/cmake/modules/AddOCaml.cmake
 @@ -70,9 +70,16 @@ function(add_ocaml_library name)
@@ -17,8 +27,11 @@ index 554046b20..b27cbd36c 100644
 +      endforeach()
 +    endif()
 +    if(LLVM_OCAML_OUT_OF_TREE)
-+      list(APPEND ocaml_flags "-ccopt" "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
++      list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
 +    endif()
  
      get_property(system_libs TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
      foreach(system_lib ${system_libs})
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.10.0.0/files/install.sh
+++ b/packages/llvm/llvm.10.0.0/files/install.sh
@@ -4,14 +4,15 @@ llvm_config="$1"
 libdir="$2"
 cmake="$3"
 make="$4"
+action="$5"
 
 function filter_experimental_targets {
     sed 's/AVR//g' | sed 's/Nios2//g' | xargs
 }
 
-function llvm_install {
-    mkdir build
-    cd build
+function llvm_build {
+    mkdir "build-$1"
+    cd "build-$1"
 
     "$cmake" \
         -DCMAKE_BUILD_TYPE="`"$llvm_config" --build-mode`" \
@@ -20,8 +21,17 @@ function llvm_install {
         -DBUILD_SHARED_LIBS=`[ $1 = "shared" ] && echo TRUE || echo FALSE` \
         -DLLVM_OCAML_OUT_OF_TREE=TRUE \
         -DLLVM_OCAML_INSTALL_PATH="${libdir}" \
+        -DLLVM_OCAML_EXTERNAL_LLVM_LIBS="`"$llvm_config" --libs`" \
         ..
     $make ocaml_all
+
+    cd ..
+}
+
+function llvm_install {
+  if test -d "build-$1"; then
+    cd "build-$1"
+
     "$cmake" -P bindings/ocaml/cmake_install.cmake
 
     mkdir "${libdir}"/llvm/$1
@@ -30,20 +40,32 @@ function llvm_install {
     mv "${libdir}"/llvm/llvm*.a "${libdir}"/llvm/$1
 
     cd ..
-    rm -rf build
+  fi
 }
 
-if "$llvm_config" --link-static --libs && "$llvm_config" --link-shared --libs; then
+case "$action" in
+"build")
+  if "$llvm_config" --link-static --libs && "$llvm_config" --link-shared --libs; then
     patch -p1 < META.patch
-    llvm_install static
-    llvm_install shared
-elif "$llvm_config" --link-static --libs; then
+    llvm_build static
+    llvm_build shared
+  elif "$llvm_config" --link-static --libs; then
     sed "s,%%LINKAGE%%,static," link-META.patch | patch -p1
-    llvm_install static
-elif "$llvm_config" --link-shared --libs; then
+    llvm_build static
+  elif "$llvm_config" --link-shared --libs; then
     sed "s,%%LINKAGE%%,shared," link-META.patch | patch -p1
-    llvm_install shared
-else
-    echo "WTF..."
+    llvm_build shared
+  else
+    echo "Link mode not recognized"
     exit 1
-fi
+  fi
+  ;;
+"install")
+  llvm_install static
+  llvm_install shared
+  ;;
+*)
+  echo "Action not recognized"
+  exit 1
+  ;;
+esac

--- a/packages/llvm/llvm.10.0.0/opam
+++ b/packages/llvm/llvm.10.0.0/opam
@@ -9,9 +9,8 @@ doc: "http://llvm.moe/ocaml"
 bug-reports: "http://llvm.org/bugs/"
 dev-repo: "git+http://llvm.org/git/llvm.git"
 homepage: "http://llvm.moe"
-install: [
-  ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make]
-]
+build: ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make "build"]
+install: ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make "install"]
 depends: [
   "ocaml" {>= "4.00.0"}
   "ctypes" {>= "0.4"}
@@ -21,16 +20,24 @@ depends: [
   "conf-python-2-7" {build}
   "conf-cmake" {build}
 ]
+conflicts: [
+  "ocaml-option-nnp"
+  "ocaml-option-nnpchecker"
+]
 patches: [
   "fix-shared.patch"
+  "fix-macos.patch"
+  "fix-rhel.patch"
 ]
 synopsis: "The OCaml bindings distributed with LLVM"
 description: "Note: LLVM should be installed first."
 extra-files: [
   ["link-META.patch" "md5=ef4ebb8706be2ed402f31fc351d7dc75"]
-  ["install.sh" "md5=76fb00c644110c73d04b15a2afb1aaf3"]
-  ["fix-shared.patch" "md5=dce86b1db352332968ceb6d042b408a8"]
+  ["install.sh" "md5=dcd03eb30e87129dbcc70a131aabfdaa"]
+  ["fix-shared.patch" "md5=818e192b6b54c0e392f2cdb2106b2e6f"]
   ["META.patch" "md5=1d0af08bab7a0f831f68849b6556e414"]
+  ["fix-macos.patch" "md5=ff33b723048e1f1a2d26505f272c8ae9"]
+  ["fix-rhel.patch" "md5=ba4142dbfc65a4b1a09ee7233048c0bf"]
 ]
 url {
   src: "https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/llvm-10.0.0.src.tar.xz"

--- a/packages/llvm/llvm.11.0.0/files/fix-macos.patch
+++ b/packages/llvm/llvm.11.0.0/files/fix-macos.patch
@@ -1,0 +1,30 @@
+From 98ca3c8802d0ae92a5baf34ffdc454d680276845 Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sat, 25 Dec 2021 18:21:57 +0000
+Subject: [PATCH 2/2] Fix support for building the OCaml binding on macOS
+
+The output of ocamlmklib is always .a and .so
+---
+ llvm/cmake/modules/AddOCaml.cmake | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index b7e2ba430344..dc80abcc6e75 100644
+--- a/cmake/modules/AddOCaml.cmake
++++ b/cmake/modules/AddOCaml.cmake
+@@ -40,10 +40,10 @@ function(add_ocaml_library name)
+   set(ocaml_outputs "${bin}/${name}.cma")
+   if( ARG_C )
+     list(APPEND ocaml_outputs
+-         "${bin}/lib${name}${CMAKE_STATIC_LIBRARY_SUFFIX}")
++         "${bin}/lib${name}.a")
+     if ( BUILD_SHARED_LIBS )
+       list(APPEND ocaml_outputs
+-           "${bin}/dll${name}${CMAKE_SHARED_LIBRARY_SUFFIX}")
++           "${bin}/dll${name}.so")
+     endif()
+   endif()
+   if( HAVE_OCAMLOPT )
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.11.0.0/files/fix-rhel.patch
+++ b/packages/llvm/llvm.11.0.0/files/fix-rhel.patch
@@ -1,0 +1,45 @@
+From 58e8402f9000d0dfe67b3df99ed8cd21b8325acd Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sun, 26 Dec 2021 05:09:10 +0000
+Subject: [PATCH] Handle RHEL-based distributions (do not have
+ libLLVM-<VERSION>.so, only libLLVM.so)
+
+---
+ llvm/cmake/modules/AddOCaml.cmake | 12 +++++++-----
+ 1 file changed, 7 insertions(+), 5 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index dc80abcc6e75..f51a339f084b 100644
+--- a/cmake/modules/AddOCaml.cmake
++++ b/cmake/modules/AddOCaml.cmake
+@@ -66,20 +66,22 @@ function(add_ocaml_library name)
+     list(APPEND ocaml_flags "-custom")
+   endif()
+ 
++  if(LLVM_OCAML_OUT_OF_TREE)
++    list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
++  endif()
++
+   if(LLVM_LINK_LLVM_DYLIB)
+     list(APPEND ocaml_flags "-lLLVM")
+   else()
+     explicit_map_components_to_libraries(llvm_libs ${ARG_LLVM})
+-    if( BUILD_SHARED_LIBS )
+-      list(APPEND ocaml_flags "-lLLVM-${LLVM_VERSION_MAJOR}${LLVM_VERSION_SUFFIX}")
++
++    if(BUILD_SHARED_LIBS AND LLVM_OCAML_OUT_OF_TREE)
++      list(APPEND ocaml_flags ${LLVM_OCAML_EXTERNAL_LLVM_LIBS})
+     else()
+       foreach( llvm_lib ${llvm_libs} )
+         list(APPEND ocaml_flags "-l${llvm_lib}" )
+       endforeach()
+     endif()
+-    if(LLVM_OCAML_OUT_OF_TREE)
+-      list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
+-    endif()
+ 
+     get_property(system_libs TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
+     foreach(system_lib ${system_libs})
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.11.0.0/files/fix-shared.patch
+++ b/packages/llvm/llvm.11.0.0/files/fix-shared.patch
@@ -1,5 +1,15 @@
-diff --git a/cmake/modules/AddOCaml.cmake b/cmake/modules/AddOCaml.cmake
-index 554046b20..b27cbd36c 100644
+From 7eaf48ca7f0b2ba0b5871b48fbe4a1b1460afbef Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sat, 25 Dec 2021 18:21:00 +0000
+Subject: [PATCH 1/2] Add support for out-of-tree shared linked OCaml LLVM
+ binding
+
+---
+ llvm/cmake/modules/AddOCaml.cmake | 13 ++++++++++---
+ 1 file changed, 10 insertions(+), 3 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index 891c9e6d618c..b7e2ba430344 100644
 --- a/cmake/modules/AddOCaml.cmake
 +++ b/cmake/modules/AddOCaml.cmake
 @@ -70,9 +70,16 @@ function(add_ocaml_library name)
@@ -17,8 +27,11 @@ index 554046b20..b27cbd36c 100644
 +      endforeach()
 +    endif()
 +    if(LLVM_OCAML_OUT_OF_TREE)
-+      list(APPEND ocaml_flags "-ccopt" "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
++      list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
 +    endif()
  
      get_property(system_libs TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
      foreach(system_lib ${system_libs})
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.11.0.0/files/install.sh
+++ b/packages/llvm/llvm.11.0.0/files/install.sh
@@ -20,6 +20,7 @@ function llvm_install {
         -DBUILD_SHARED_LIBS=`[ $1 = "shared" ] && echo TRUE || echo FALSE` \
         -DLLVM_OCAML_OUT_OF_TREE=TRUE \
         -DLLVM_OCAML_INSTALL_PATH="${libdir}" \
+        -DLLVM_OCAML_EXTERNAL_LLVM_LIBS="`"$llvm_config" --libs`" \
         ..
     $make ocaml_all
     "$cmake" -P bindings/ocaml/cmake_install.cmake

--- a/packages/llvm/llvm.11.0.0/files/install.sh
+++ b/packages/llvm/llvm.11.0.0/files/install.sh
@@ -4,14 +4,15 @@ llvm_config="$1"
 libdir="$2"
 cmake="$3"
 make="$4"
+action="$5"
 
 function filter_experimental_targets {
     sed 's/AVR//g' | sed 's/Nios2//g' | xargs
 }
 
-function llvm_install {
-    mkdir build
-    cd build
+function llvm_build {
+    mkdir "build-$1"
+    cd "build-$1"
 
     "$cmake" \
         -DCMAKE_BUILD_TYPE="`"$llvm_config" --build-mode`" \
@@ -23,6 +24,14 @@ function llvm_install {
         -DLLVM_OCAML_EXTERNAL_LLVM_LIBS="`"$llvm_config" --libs`" \
         ..
     $make ocaml_all
+
+    cd ..
+}
+
+function llvm_install {
+  if test -d "build-$1"; then
+    cd "build-$1"
+
     "$cmake" -P bindings/ocaml/cmake_install.cmake
 
     mkdir "${libdir}"/llvm/$1
@@ -31,20 +40,32 @@ function llvm_install {
     mv "${libdir}"/llvm/llvm*.a "${libdir}"/llvm/$1
 
     cd ..
-    rm -rf build
+  fi
 }
 
-if "$llvm_config" --link-static --libs && "$llvm_config" --link-shared --libs; then
+case "$action" in
+"build")
+  if "$llvm_config" --link-static --libs && "$llvm_config" --link-shared --libs; then
     patch -p1 < META.patch
-    llvm_install static
-    llvm_install shared
-elif "$llvm_config" --link-static --libs; then
+    llvm_build static
+    llvm_build shared
+  elif "$llvm_config" --link-static --libs; then
     sed "s,%%LINKAGE%%,static," link-META.patch | patch -p1
-    llvm_install static
-elif "$llvm_config" --link-shared --libs; then
+    llvm_build static
+  elif "$llvm_config" --link-shared --libs; then
     sed "s,%%LINKAGE%%,shared," link-META.patch | patch -p1
-    llvm_install shared
-else
-    echo "WTF..."
+    llvm_build shared
+  else
+    echo "Link mode not recognized"
     exit 1
-fi
+  fi
+  ;;
+"install")
+  llvm_install static
+  llvm_install shared
+  ;;
+*)
+  echo "Action not recognized"
+  exit 1
+  ;;
+esac

--- a/packages/llvm/llvm.11.0.0/opam
+++ b/packages/llvm/llvm.11.0.0/opam
@@ -23,14 +23,16 @@ depends: [
 ]
 patches: [
   "fix-shared.patch"
+  "fix-macos.patch"
 ]
 synopsis: "The OCaml bindings distributed with LLVM"
 description: "Note: LLVM should be installed first."
 extra-files: [
   ["link-META.patch" "md5=ef4ebb8706be2ed402f31fc351d7dc75"]
   ["install.sh" "md5=76fb00c644110c73d04b15a2afb1aaf3"]
-  ["fix-shared.patch" "md5=dce86b1db352332968ceb6d042b408a8"]
+  ["fix-shared.patch" "md5=818e192b6b54c0e392f2cdb2106b2e6f"]
   ["META.patch" "md5=1d0af08bab7a0f831f68849b6556e414"]
+  ["fix-macos.patch" "md5=ff33b723048e1f1a2d26505f272c8ae9"]
 ]
 url {
   src: "https://github.com/llvm/llvm-project/releases/download/llvmorg-11.1.0/llvm-11.1.0.src.tar.xz"

--- a/packages/llvm/llvm.11.0.0/opam
+++ b/packages/llvm/llvm.11.0.0/opam
@@ -29,7 +29,7 @@ synopsis: "The OCaml bindings distributed with LLVM"
 description: "Note: LLVM should be installed first."
 extra-files: [
   ["link-META.patch" "md5=ef4ebb8706be2ed402f31fc351d7dc75"]
-  ["install.sh" "md5=76fb00c644110c73d04b15a2afb1aaf3"]
+  ["install.sh" "md5=dcd03eb30e87129dbcc70a131aabfdaa"]
   ["fix-shared.patch" "md5=818e192b6b54c0e392f2cdb2106b2e6f"]
   ["META.patch" "md5=1d0af08bab7a0f831f68849b6556e414"]
   ["fix-macos.patch" "md5=ff33b723048e1f1a2d26505f272c8ae9"]

--- a/packages/llvm/llvm.11.0.0/opam
+++ b/packages/llvm/llvm.11.0.0/opam
@@ -9,9 +9,8 @@ doc: "http://llvm.moe/ocaml"
 bug-reports: "http://llvm.org/bugs/"
 dev-repo: "git+http://llvm.org/git/llvm.git"
 homepage: "http://llvm.moe"
-install: [
-  ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make]
-]
+build: ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make "build"]
+install: ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make "install"]
 depends: [
   "ocaml" {>= "4.00.0"}
   "ctypes" {>= "0.4"}

--- a/packages/llvm/llvm.11.0.0/opam
+++ b/packages/llvm/llvm.11.0.0/opam
@@ -20,6 +20,10 @@ depends: [
   "conf-python-2-7" {build}
   "conf-cmake" {build}
 ]
+conflicts: [
+  "ocaml-option-nnp"
+  "ocaml-option-nnpchecker"
+]
 patches: [
   "fix-shared.patch"
   "fix-macos.patch"

--- a/packages/llvm/llvm.11.0.0/opam
+++ b/packages/llvm/llvm.11.0.0/opam
@@ -24,6 +24,7 @@ depends: [
 patches: [
   "fix-shared.patch"
   "fix-macos.patch"
+  "fix-rhel.patch"
 ]
 synopsis: "The OCaml bindings distributed with LLVM"
 description: "Note: LLVM should be installed first."
@@ -33,6 +34,7 @@ extra-files: [
   ["fix-shared.patch" "md5=818e192b6b54c0e392f2cdb2106b2e6f"]
   ["META.patch" "md5=1d0af08bab7a0f831f68849b6556e414"]
   ["fix-macos.patch" "md5=ff33b723048e1f1a2d26505f272c8ae9"]
+  ["fix-rhel.patch" "md5=ba4142dbfc65a4b1a09ee7233048c0bf"]
 ]
 url {
   src: "https://github.com/llvm/llvm-project/releases/download/llvmorg-11.1.0/llvm-11.1.0.src.tar.xz"

--- a/packages/llvm/llvm.12.0.1/files/fix-macos.patch
+++ b/packages/llvm/llvm.12.0.1/files/fix-macos.patch
@@ -1,0 +1,30 @@
+From 98ca3c8802d0ae92a5baf34ffdc454d680276845 Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sat, 25 Dec 2021 18:21:57 +0000
+Subject: [PATCH 2/2] Fix support for building the OCaml binding on macOS
+
+The output of ocamlmklib is always .a and .so
+---
+ llvm/cmake/modules/AddOCaml.cmake | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index b7e2ba430344..dc80abcc6e75 100644
+--- a/cmake/modules/AddOCaml.cmake
++++ b/cmake/modules/AddOCaml.cmake
+@@ -40,10 +40,10 @@ function(add_ocaml_library name)
+   set(ocaml_outputs "${bin}/${name}.cma")
+   if( ARG_C )
+     list(APPEND ocaml_outputs
+-         "${bin}/lib${name}${CMAKE_STATIC_LIBRARY_SUFFIX}")
++         "${bin}/lib${name}.a")
+     if ( BUILD_SHARED_LIBS )
+       list(APPEND ocaml_outputs
+-           "${bin}/dll${name}${CMAKE_SHARED_LIBRARY_SUFFIX}")
++           "${bin}/dll${name}.so")
+     endif()
+   endif()
+   if( HAVE_OCAMLOPT )
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.12.0.1/files/fix-rhel.patch
+++ b/packages/llvm/llvm.12.0.1/files/fix-rhel.patch
@@ -1,0 +1,45 @@
+From 58e8402f9000d0dfe67b3df99ed8cd21b8325acd Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sun, 26 Dec 2021 05:09:10 +0000
+Subject: [PATCH] Handle RHEL-based distributions (do not have
+ libLLVM-<VERSION>.so, only libLLVM.so)
+
+---
+ llvm/cmake/modules/AddOCaml.cmake | 12 +++++++-----
+ 1 file changed, 7 insertions(+), 5 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index dc80abcc6e75..f51a339f084b 100644
+--- a/cmake/modules/AddOCaml.cmake
++++ b/cmake/modules/AddOCaml.cmake
+@@ -66,20 +66,22 @@ function(add_ocaml_library name)
+     list(APPEND ocaml_flags "-custom")
+   endif()
+ 
++  if(LLVM_OCAML_OUT_OF_TREE)
++    list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
++  endif()
++
+   if(LLVM_LINK_LLVM_DYLIB)
+     list(APPEND ocaml_flags "-lLLVM")
+   else()
+     explicit_map_components_to_libraries(llvm_libs ${ARG_LLVM})
+-    if( BUILD_SHARED_LIBS )
+-      list(APPEND ocaml_flags "-lLLVM-${LLVM_VERSION_MAJOR}${LLVM_VERSION_SUFFIX}")
++
++    if(BUILD_SHARED_LIBS AND LLVM_OCAML_OUT_OF_TREE)
++      list(APPEND ocaml_flags ${LLVM_OCAML_EXTERNAL_LLVM_LIBS})
+     else()
+       foreach( llvm_lib ${llvm_libs} )
+         list(APPEND ocaml_flags "-l${llvm_lib}" )
+       endforeach()
+     endif()
+-    if(LLVM_OCAML_OUT_OF_TREE)
+-      list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
+-    endif()
+ 
+     get_property(system_libs TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
+     foreach(system_lib ${system_libs})
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.12.0.1/files/fix-shared.patch
+++ b/packages/llvm/llvm.12.0.1/files/fix-shared.patch
@@ -1,5 +1,15 @@
-diff --git a/cmake/modules/AddOCaml.cmake b/cmake/modules/AddOCaml.cmake
-index 554046b20..b27cbd36c 100644
+From 7eaf48ca7f0b2ba0b5871b48fbe4a1b1460afbef Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sat, 25 Dec 2021 18:21:00 +0000
+Subject: [PATCH 1/2] Add support for out-of-tree shared linked OCaml LLVM
+ binding
+
+---
+ llvm/cmake/modules/AddOCaml.cmake | 13 ++++++++++---
+ 1 file changed, 10 insertions(+), 3 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index 891c9e6d618c..b7e2ba430344 100644
 --- a/cmake/modules/AddOCaml.cmake
 +++ b/cmake/modules/AddOCaml.cmake
 @@ -70,9 +70,16 @@ function(add_ocaml_library name)
@@ -17,8 +27,11 @@ index 554046b20..b27cbd36c 100644
 +      endforeach()
 +    endif()
 +    if(LLVM_OCAML_OUT_OF_TREE)
-+      list(APPEND ocaml_flags "-ccopt" "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
++      list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
 +    endif()
  
      get_property(system_libs TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
      foreach(system_lib ${system_libs})
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.12.0.1/files/install.sh
+++ b/packages/llvm/llvm.12.0.1/files/install.sh
@@ -4,14 +4,15 @@ llvm_config="$1"
 libdir="$2"
 cmake="$3"
 make="$4"
+action="$5"
 
 function filter_experimental_targets {
     sed 's/AVR//g' | sed 's/Nios2//g' | xargs
 }
 
-function llvm_install {
-    mkdir build
-    cd build
+function llvm_build {
+    mkdir "build-$1"
+    cd "build-$1"
 
     "$cmake" \
         -DCMAKE_BUILD_TYPE="`"$llvm_config" --build-mode`" \
@@ -20,8 +21,17 @@ function llvm_install {
         -DBUILD_SHARED_LIBS=`[ $1 = "shared" ] && echo TRUE || echo FALSE` \
         -DLLVM_OCAML_OUT_OF_TREE=TRUE \
         -DLLVM_OCAML_INSTALL_PATH="${libdir}" \
+        -DLLVM_OCAML_EXTERNAL_LLVM_LIBS="`"$llvm_config" --libs`" \
         ..
     $make ocaml_all
+
+    cd ..
+}
+
+function llvm_install {
+  if test -d "build-$1"; then
+    cd "build-$1"
+
     "$cmake" -P bindings/ocaml/cmake_install.cmake
 
     mkdir "${libdir}"/llvm/$1
@@ -30,20 +40,32 @@ function llvm_install {
     mv "${libdir}"/llvm/llvm*.a "${libdir}"/llvm/$1
 
     cd ..
-    rm -rf build
+  fi
 }
 
-if "$llvm_config" --link-static --libs && "$llvm_config" --link-shared --libs; then
+case "$action" in
+"build")
+  if "$llvm_config" --link-static --libs && "$llvm_config" --link-shared --libs; then
     patch -p1 < META.patch
-    llvm_install static
-    llvm_install shared
-elif "$llvm_config" --link-static --libs; then
+    llvm_build static
+    llvm_build shared
+  elif "$llvm_config" --link-static --libs; then
     sed "s,%%LINKAGE%%,static," link-META.patch | patch -p1
-    llvm_install static
-elif "$llvm_config" --link-shared --libs; then
+    llvm_build static
+  elif "$llvm_config" --link-shared --libs; then
     sed "s,%%LINKAGE%%,shared," link-META.patch | patch -p1
-    llvm_install shared
-else
-    echo "WTF..."
+    llvm_build shared
+  else
+    echo "Link mode not recognized"
     exit 1
-fi
+  fi
+  ;;
+"install")
+  llvm_install static
+  llvm_install shared
+  ;;
+*)
+  echo "Action not recognized"
+  exit 1
+  ;;
+esac

--- a/packages/llvm/llvm.12.0.1/opam
+++ b/packages/llvm/llvm.12.0.1/opam
@@ -9,9 +9,8 @@ doc: "http://llvm.moe/ocaml"
 bug-reports: "http://llvm.org/bugs/"
 dev-repo: "git+http://llvm.org/git/llvm.git"
 homepage: "http://llvm.moe"
-install: [
-  ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make]
-]
+build: ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make "build"]
+install: ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make "install"]
 depends: [
   "ocaml" {>= "4.00.0"}
   "ctypes" {>= "0.4"}
@@ -21,16 +20,24 @@ depends: [
   "conf-python-2-7" {build}
   "conf-cmake" {build}
 ]
+conflicts: [
+  "ocaml-option-nnp"
+  "ocaml-option-nnpchecker"
+]
 patches: [
   "fix-shared.patch"
+  "fix-macos.patch"
+  "fix-rhel.patch"
 ]
 synopsis: "The OCaml bindings distributed with LLVM"
 description: "Note: LLVM should be installed first."
 extra-files: [
   ["link-META.patch" "md5=ef4ebb8706be2ed402f31fc351d7dc75"]
-  ["install.sh" "md5=76fb00c644110c73d04b15a2afb1aaf3"]
-  ["fix-shared.patch" "md5=dce86b1db352332968ceb6d042b408a8"]
+  ["install.sh" "md5=dcd03eb30e87129dbcc70a131aabfdaa"]
+  ["fix-shared.patch" "md5=818e192b6b54c0e392f2cdb2106b2e6f"]
   ["META.patch" "md5=1d0af08bab7a0f831f68849b6556e414"]
+  ["fix-macos.patch" "md5=ff33b723048e1f1a2d26505f272c8ae9"]
+  ["fix-rhel.patch" "md5=ba4142dbfc65a4b1a09ee7233048c0bf"]
 ]
 url {
   src: "https://github.com/llvm/llvm-project/releases/download/llvmorg-12.0.1/llvm-12.0.1.src.tar.xz"

--- a/packages/llvm/llvm.13.0.0/files/fix-macos.patch
+++ b/packages/llvm/llvm.13.0.0/files/fix-macos.patch
@@ -1,0 +1,30 @@
+From 98ca3c8802d0ae92a5baf34ffdc454d680276845 Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sat, 25 Dec 2021 18:21:57 +0000
+Subject: [PATCH 2/2] Fix support for building the OCaml binding on macOS
+
+The output of ocamlmklib is always .a and .so
+---
+ llvm/cmake/modules/AddOCaml.cmake | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index b7e2ba430344..dc80abcc6e75 100644
+--- a/cmake/modules/AddOCaml.cmake
++++ b/cmake/modules/AddOCaml.cmake
+@@ -40,10 +40,10 @@ function(add_ocaml_library name)
+   set(ocaml_outputs "${bin}/${name}.cma")
+   if( ARG_C )
+     list(APPEND ocaml_outputs
+-         "${bin}/lib${name}${CMAKE_STATIC_LIBRARY_SUFFIX}")
++         "${bin}/lib${name}.a")
+     if ( BUILD_SHARED_LIBS )
+       list(APPEND ocaml_outputs
+-           "${bin}/dll${name}${CMAKE_SHARED_LIBRARY_SUFFIX}")
++           "${bin}/dll${name}.so")
+     endif()
+   endif()
+   if( HAVE_OCAMLOPT )
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.13.0.0/files/fix-rhel.patch
+++ b/packages/llvm/llvm.13.0.0/files/fix-rhel.patch
@@ -1,0 +1,45 @@
+From 58e8402f9000d0dfe67b3df99ed8cd21b8325acd Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sun, 26 Dec 2021 05:09:10 +0000
+Subject: [PATCH] Handle RHEL-based distributions (do not have
+ libLLVM-<VERSION>.so, only libLLVM.so)
+
+---
+ llvm/cmake/modules/AddOCaml.cmake | 12 +++++++-----
+ 1 file changed, 7 insertions(+), 5 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index dc80abcc6e75..f51a339f084b 100644
+--- a/cmake/modules/AddOCaml.cmake
++++ b/cmake/modules/AddOCaml.cmake
+@@ -66,20 +66,22 @@ function(add_ocaml_library name)
+     list(APPEND ocaml_flags "-custom")
+   endif()
+ 
++  if(LLVM_OCAML_OUT_OF_TREE)
++    list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
++  endif()
++
+   if(LLVM_LINK_LLVM_DYLIB)
+     list(APPEND ocaml_flags "-lLLVM")
+   else()
+     explicit_map_components_to_libraries(llvm_libs ${ARG_LLVM})
+-    if( BUILD_SHARED_LIBS )
+-      list(APPEND ocaml_flags "-lLLVM-${LLVM_VERSION_MAJOR}${LLVM_VERSION_SUFFIX}")
++
++    if(BUILD_SHARED_LIBS AND LLVM_OCAML_OUT_OF_TREE)
++      list(APPEND ocaml_flags ${LLVM_OCAML_EXTERNAL_LLVM_LIBS})
+     else()
+       foreach( llvm_lib ${llvm_libs} )
+         list(APPEND ocaml_flags "-l${llvm_lib}" )
+       endforeach()
+     endif()
+-    if(LLVM_OCAML_OUT_OF_TREE)
+-      list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
+-    endif()
+ 
+     get_property(system_libs TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
+     foreach(system_lib ${system_libs})
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.13.0.0/files/fix-shared.patch
+++ b/packages/llvm/llvm.13.0.0/files/fix-shared.patch
@@ -1,5 +1,15 @@
-diff --git a/cmake/modules/AddOCaml.cmake b/cmake/modules/AddOCaml.cmake
-index 554046b20..b27cbd36c 100644
+From 7eaf48ca7f0b2ba0b5871b48fbe4a1b1460afbef Mon Sep 17 00:00:00 2001
+From: Kate <kit.ty.kate@disroot.org>
+Date: Sat, 25 Dec 2021 18:21:00 +0000
+Subject: [PATCH 1/2] Add support for out-of-tree shared linked OCaml LLVM
+ binding
+
+---
+ llvm/cmake/modules/AddOCaml.cmake | 13 ++++++++++---
+ 1 file changed, 10 insertions(+), 3 deletions(-)
+
+diff --git a/llvm/cmake/modules/AddOCaml.cmake b/llvm/cmake/modules/AddOCaml.cmake
+index 891c9e6d618c..b7e2ba430344 100644
 --- a/cmake/modules/AddOCaml.cmake
 +++ b/cmake/modules/AddOCaml.cmake
 @@ -70,9 +70,16 @@ function(add_ocaml_library name)
@@ -17,8 +27,11 @@ index 554046b20..b27cbd36c 100644
 +      endforeach()
 +    endif()
 +    if(LLVM_OCAML_OUT_OF_TREE)
-+      list(APPEND ocaml_flags "-ccopt" "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
++      list(APPEND ocaml_flags "-L${LLVM_OCAML_EXTERNAL_LLVM_LIBDIR}" )
 +    endif()
  
      get_property(system_libs TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
      foreach(system_lib ${system_libs})
+-- 
+2.34.0
+

--- a/packages/llvm/llvm.13.0.0/files/install.sh
+++ b/packages/llvm/llvm.13.0.0/files/install.sh
@@ -4,14 +4,15 @@ llvm_config="$1"
 libdir="$2"
 cmake="$3"
 make="$4"
+action="$5"
 
 function filter_experimental_targets {
-    sed 's/AVR//g' | sed 's/M68k//g' | sed 's/Nios2//g' | xargs
+    sed 's/AVR//g' | sed 's/Nios2//g' | xargs
 }
 
-function llvm_install {
-    mkdir build
-    cd build
+function llvm_build {
+    mkdir "build-$1"
+    cd "build-$1"
 
     "$cmake" \
         -DCMAKE_BUILD_TYPE="`"$llvm_config" --build-mode`" \
@@ -20,8 +21,17 @@ function llvm_install {
         -DBUILD_SHARED_LIBS=`[ $1 = "shared" ] && echo TRUE || echo FALSE` \
         -DLLVM_OCAML_OUT_OF_TREE=TRUE \
         -DLLVM_OCAML_INSTALL_PATH="${libdir}" \
+        -DLLVM_OCAML_EXTERNAL_LLVM_LIBS="`"$llvm_config" --libs`" \
         ..
     $make ocaml_all
+
+    cd ..
+}
+
+function llvm_install {
+  if test -d "build-$1"; then
+    cd "build-$1"
+
     "$cmake" -P bindings/ocaml/cmake_install.cmake
 
     mkdir "${libdir}"/llvm/$1
@@ -30,20 +40,32 @@ function llvm_install {
     mv "${libdir}"/llvm/llvm*.a "${libdir}"/llvm/$1
 
     cd ..
-    rm -rf build
+  fi
 }
 
-if "$llvm_config" --link-static --libs && "$llvm_config" --link-shared --libs; then
+case "$action" in
+"build")
+  if "$llvm_config" --link-static --libs && "$llvm_config" --link-shared --libs; then
     patch -p1 < META.patch
-    llvm_install static
-    llvm_install shared
-elif "$llvm_config" --link-static --libs; then
+    llvm_build static
+    llvm_build shared
+  elif "$llvm_config" --link-static --libs; then
     sed "s,%%LINKAGE%%,static," link-META.patch | patch -p1
-    llvm_install static
-elif "$llvm_config" --link-shared --libs; then
+    llvm_build static
+  elif "$llvm_config" --link-shared --libs; then
     sed "s,%%LINKAGE%%,shared," link-META.patch | patch -p1
-    llvm_install shared
-else
-    echo "WTF..."
+    llvm_build shared
+  else
+    echo "Link mode not recognized"
     exit 1
-fi
+  fi
+  ;;
+"install")
+  llvm_install static
+  llvm_install shared
+  ;;
+*)
+  echo "Action not recognized"
+  exit 1
+  ;;
+esac

--- a/packages/llvm/llvm.13.0.0/files/install.sh
+++ b/packages/llvm/llvm.13.0.0/files/install.sh
@@ -7,7 +7,7 @@ make="$4"
 action="$5"
 
 function filter_experimental_targets {
-    sed 's/AVR//g' | sed 's/Nios2//g' | xargs
+    sed 's/AVR//g' | sed 's/M68k//g' | sed 's/Nios2//g' | xargs
 }
 
 function llvm_build {

--- a/packages/llvm/llvm.13.0.0/opam
+++ b/packages/llvm/llvm.13.0.0/opam
@@ -33,7 +33,7 @@ patches: [
 synopsis: "The OCaml bindings distributed with LLVM"
 description: "Note: LLVM should be installed first."
 extra-files: [
-  ["link-META.patch" "md5=ef4ebb8706be2ed402f31fc351d7dc75"]
+  ["link-META.patch" "md5=53dc8e72917585d235b9bbb3e06eed2b"]
   ["install.sh" "md5=dcd03eb30e87129dbcc70a131aabfdaa"]
   ["fix-shared.patch" "md5=818e192b6b54c0e392f2cdb2106b2e6f"]
   ["add-cmt-for-llvm-13.patch" "md5=248ec7d17ebd1592fc99848016d9452b"]

--- a/packages/llvm/llvm.13.0.0/opam
+++ b/packages/llvm/llvm.13.0.0/opam
@@ -34,7 +34,7 @@ synopsis: "The OCaml bindings distributed with LLVM"
 description: "Note: LLVM should be installed first."
 extra-files: [
   ["link-META.patch" "md5=53dc8e72917585d235b9bbb3e06eed2b"]
-  ["install.sh" "md5=dcd03eb30e87129dbcc70a131aabfdaa"]
+  ["install.sh" "md5=f318260916581cbf0a8e7aa2903ac6d2"]
   ["fix-shared.patch" "md5=818e192b6b54c0e392f2cdb2106b2e6f"]
   ["add-cmt-for-llvm-13.patch" "md5=248ec7d17ebd1592fc99848016d9452b"]
   ["META.patch" "md5=4c3cf99d06d2044e2ce5a853e5a16d5e"]

--- a/packages/llvm/llvm.13.0.0/opam
+++ b/packages/llvm/llvm.13.0.0/opam
@@ -24,15 +24,17 @@ depends: [
 patches: [
   "fix-shared.patch"
   "add-cmt-for-llvm-13.patch"
+  "fix-macos.patch"
 ]
 synopsis: "The OCaml bindings distributed with LLVM"
 description: "Note: LLVM should be installed first."
 extra-files: [
   ["link-META.patch" "md5=53dc8e72917585d235b9bbb3e06eed2b"]
   ["install.sh" "md5=fb51c3d8bc987acac39927c2abd84bcd"]
-  ["fix-shared.patch" "md5=dce86b1db352332968ceb6d042b408a8"]
+  ["fix-shared.patch" "md5=818e192b6b54c0e392f2cdb2106b2e6f"]
   ["add-cmt-for-llvm-13.patch" "md5=248ec7d17ebd1592fc99848016d9452b"]
   ["META.patch" "md5=4c3cf99d06d2044e2ce5a853e5a16d5e"]
+  ["fix-macos.patch" "md5=ff33b723048e1f1a2d26505f272c8ae9"]
 ]
 url {
   src: "https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/llvm-13.0.0.src.tar.xz"

--- a/packages/llvm/llvm.13.0.0/opam
+++ b/packages/llvm/llvm.13.0.0/opam
@@ -9,9 +9,8 @@ doc: "http://llvm.moe/ocaml"
 bug-reports: "http://llvm.org/bugs/"
 dev-repo: "git+http://llvm.org/git/llvm.git"
 homepage: "http://llvm.moe"
-install: [
-  ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make]
-]
+build: ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make "build"]
+install: ["bash" "-ex" "install.sh" "%{conf-llvm:config}%" lib "%{conf-cmake:cmd}%" make "install"]
 depends: [
   "ocaml" {>= "4.00.0"}
   "ctypes" {>= "0.4"}
@@ -21,20 +20,26 @@ depends: [
   "conf-python-3" {build}
   "conf-cmake" {build}
 ]
+conflicts: [
+  "ocaml-option-nnp"
+  "ocaml-option-nnpchecker"
+]
 patches: [
   "fix-shared.patch"
   "add-cmt-for-llvm-13.patch"
   "fix-macos.patch"
+  "fix-rhel.patch"
 ]
 synopsis: "The OCaml bindings distributed with LLVM"
 description: "Note: LLVM should be installed first."
 extra-files: [
-  ["link-META.patch" "md5=53dc8e72917585d235b9bbb3e06eed2b"]
-  ["install.sh" "md5=fb51c3d8bc987acac39927c2abd84bcd"]
+  ["link-META.patch" "md5=ef4ebb8706be2ed402f31fc351d7dc75"]
+  ["install.sh" "md5=dcd03eb30e87129dbcc70a131aabfdaa"]
   ["fix-shared.patch" "md5=818e192b6b54c0e392f2cdb2106b2e6f"]
   ["add-cmt-for-llvm-13.patch" "md5=248ec7d17ebd1592fc99848016d9452b"]
   ["META.patch" "md5=4c3cf99d06d2044e2ce5a853e5a16d5e"]
   ["fix-macos.patch" "md5=ff33b723048e1f1a2d26505f272c8ae9"]
+  ["fix-rhel.patch" "md5=ba4142dbfc65a4b1a09ee7233048c0bf"]
 ]
 url {
   src: "https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/llvm-13.0.0.src.tar.xz"

